File: c/src/kill.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/types.h>
#include <signal.h>

int main(int argc, char *argv[])
{
    int pid;

    if (argc != 2) {
        fprintf(stdout, "Usage: %s <pid>\n", argv[0]);
        exit(1);
    }

    pid = atoi(argv[1]);
    if (kill(pid, SIGKILL) < 0) {
        perror(argv[1]);
        exit(1);
    }
    exit(0);
}

File: golang/src/kill.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func main() {
	if len(os.Args) < 2 {
		fmt.Printf("Usage: %s <pid>\n", os.Args[0])
		return
	}
	for i, v := range os.Args {
		if i == 0 {
			continue
		}
		pid, err := strconv.Atoi(v);
		if err != nil {
			die(err)
		}
		doKill(pid)
	}
}

func doKill(pid int) error {
	procs, err := os.FindProcess(pid)
	if err != nil {
		die(err)
	}
	err = procs.Kill()
	if err != nil {
		die(err)
	}
	return nil
}

func die(e error) {
	fmt.Fprintf(os.Stderr, "%s\n", e.Error())
	os.Exit(1)
}
